src/main/java/com/solvd/noteservice/web/controller/NoteController.java

```diff
@@ -26,45 +29,52 @@ public class NoteController {
     private final NoteMapper noteMapper;
 
     @PostMapping
-    public final NoteDto create(@RequestBody @Validated final NoteDto noteDto) {
-        Note note = noteMapper.toEntity(noteDto);
-        note = noteService.create(note);
-        return noteMapper.toDto(note);
+    @MutationMapping
+    public final NoteDto create(@RequestBody @Validated @Argument final NoteDto note) {
+        Note noteMapped = noteMapper.toEntity(note);
+        noteMapped = noteService.create(noteMapped);
```

Comment:
@vpetrovaa why do don't create a new instance of `Note`?

Reply:
@h1alexbel  You mean
```suggestion
        Note noteSaved = noteService.create(noteMapped);
```
? But what for? We just set the id and all other fields are the same. Will it not be unnecessary?

Reply:
@vpetrovaa I think mostly because of side effects. After transformation you should return a new instance of the object

thus, side effects will be applied only on the latest instance (version) of the object
more traceable in the future maintenance of the application logic

WDYT?

Reply:
ok, It sounds convincing, https://github.com/vpetrovaa/note-service/pull/20/commits/14d3662fe3e7fdf1c5f5fe16272ec6caafedfe66